### PR TITLE
Adding median absolute deviation documentation page

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description
Adds documentation for the `median_absolute_deviation` aggregation function.

### Issues Resolved
https://github.com/opensearch-project/documentation-website/issues/4293

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
@martin-gaievski - Can you sign off on this from a technical standpoint? Thanks.

**Comment by GumpacG** [comment]
This may need more context regarding the `compression` field. https://github.com/opensearch-project/OpenSearch/blob/ef2a9e5a290262201f069539de45248e1bfe34e4/server/src/main/java/org/opensearch/search/aggregations/metrics/MedianAbsoluteDeviationAggregationBuilder.java#L106

**Comment by hdhalter** [comment]
@sandeshkr419 - Do you approve of this content from a technical standpoint? 

@vagimeli for visibility, as you are working on aggregations content.

File: _aggregations/metric/median-absolute-deviation.md
Comment (sandeshkr419) [inline _aggregations/metric/median-absolute-deviation.md:65]: Should we just rephrase this to:

```
By default, a missing field in or a null value in a field from a document are ignored in computations. You can specify a value for them by specifying the `missing` parameter. 
```

I don't like the idea of saying that we are providing a default value. It is more like treat the missing values as this value which I'm specifying.

File: _aggregations/metric/median-absolute-deviation.md
Comment (sandeshkr419) [inline _aggregations/metric/median-absolute-deviation.md:112]: I think we need to add in a little (very-little) detail on `compression` here. 
  
  
How about:

-----
The calculation of the median absolute deviation utilizes [t-digest](https://github.com/tdunning/t-digest/tree/main) data-structure which controls the balance between performance and accuracy of estimation. T-Digest controls this balance through `compression` parameter, with a default value of 1000. Adjusting the `compression` value affects the trade-off between computational efficiency and the precision of the estimation. Lowering the compression value improves performance but may lead to slightly less accurate results, while increasing it enhances accuracy at the cost of increased computational overhead.

----

**Comment by sandeshkr419** [comment]
@MitchellGale Thanks for drafting up the doc. 
I have some comments on adding more details to few sections for better understanding.

File: _aggregations/metric/median-absolute-deviation.md
Comment (vagimeli) [inline _aggregations/metric/median-absolute-deviation.md:13]:
```suggestion
The `median_absolute_deviation` metric is a single-value metric aggregation that returns median absolute deviation field. Median absolute deviation is a statistical measure of data variability. Since the median absolute deviation measures dispersion from the median, it provides a more robust measure of variability that is less affected by outliers in a dataset. 
```

File: _aggregations/metric/median-absolute-deviation.md
Comment (vagimeli) [inline _aggregations/metric/median-absolute-deviation.md:18]:
```suggestion
The following example calculates the median absolute deviation of the `DistanceMiles` field of the sample dataset `opensearch_dashboards_sample_data_flights`:
```

File: _aggregations/metric/median-absolute-deviation.md
Comment (vagimeli) [inline _aggregations/metric/median-absolute-deviation.md:65]:
```suggestion
You can set a default value for fields that are missing from documents by specifying the `missing` parameter. The value can be `missing` or `null`.
```

File: _aggregations/metric/median-absolute-deviation.md
Comment (vagimeli) [inline _aggregations/metric/median-absolute-deviation.md:112]:
```suggestion
The median absolute deviation calculation uses [t-digest](https://github.com/tdunning/t-digest/tree/main), which controls the balance between performance and accuracy of estimation. The default value for the t-digest `compression` field is `1000`. Decreasing the `compression` value increases the performance while reducing the accuracy of estimation.
```

**Review by vagimeli — APPROVED** [review]
Doc review complete. I'll move this to editorial once suggestions are addressed.

**Comment by vagimeli** [comment]
> Doc review complete. I'll move this to editorial once suggestions are addressed.

@MitchellGale @sandeshkr419 Please let me know know once the doc review changes are accepted. This will need to go through editorial review before we can merge it. I'll handle that process. We just need an SME's tech review approval before I can move this forward. Thank you.

**Review by sandeshkr419 — APPROVED** [review]
Thanks @vagimeli for these suggestive changes in this documentation. LGTM, approving it!

**Pull request closed without merging**
